native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -1,20 +1,113 @@
 import UIKit
 import SharedCode
 
-class ViewController: UIViewController {
+class TableViewCell: UITableViewCell {
+    
+    @IBOutlet weak var departureLabel: UILabel!
+    @IBOutlet weak var arrivalLabel: UILabel!
+}
 
-    @IBOutlet private var label: UILabel!
+class ViewController: UIViewController, ApplicationContractView{
+
+    @IBOutlet weak var dropDown1: UIPickerView!
+    @IBOutlet weak var dropDown2: UIPickerView!
+    @IBOutlet weak var tableView: UITableView!
+    
+    private var tableData: NSMutableArray = NSMutableArray()
+    
+    private var stations: [String] = [String]()
+    
 
     private let presenter: ApplicationContractPresenter = ApplicationPresenter()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         presenter.onViewTaken(view: self)
+        
+        setUpPickers()
+        setUpTable()
+    }
+    
+    
+    @IBAction func onButtonTapped(){
+        let originStation = stations[dropDown1.selectedRow(inComponent:0)]
+        let finalStation = stations[dropDown2.selectedRow(inComponent:0)]
+        presenter.makeTrainSearch(originCrs: originStation,destinationCrs:  finalStation)
```

Comment:
Nit: Can be have a space before `destinationCrs` and I think there's two spaces after the second colon, can we reduce this to one please